File: cloudnet-app-lib/cloudnet-logging/src/main/java/eu/cloudnetservice/cloudnet/v2/logging/LoggingFormatter.java
package eu.cloudnetservice.cloudnet.v2.logging;

import org.fusesource.jansi.Ansi;

import java.io.PrintWriter;
import java.io.StringWriter;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.logging.Formatter;
import java.util.logging.Level;
import java.util.logging.LogRecord;

public class LoggingFormatter extends Formatter {

    private final DateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss");

    @Override
    public String format(final LogRecord record) {
        StringBuilder builder = new StringBuilder();
        if (record != null) {

            colorMessage(builder, record);

            if (record.getThrown() != null) {
                final StringWriter stringWriter = new StringWriter();
                record.getThrown().printStackTrace(new PrintWriter(stringWriter));
                builder.append(stringWriter);
                builder.append(System.lineSeparator());
            }

        }
        return builder.toString();
    }

    private void colorMessage(StringBuilder builder, LogRecord record) {
        String message = formatMessage(record);
        builder.append('[');
        builder.append(dateFormat.format(record.getMillis()));
        builder.append("] ");
        builder.append(colorLevel(record));
        builder.append(": ");
        builder.append(message);
        if (!message.endsWith(System.lineSeparator())) {
            builder.append(System.lineSeparator());
        }
    }

    private String colorLevel(LogRecord logRecord) {
        if (logRecord.getLevel().equals(Level.WARNING)) {
            return Ansi.ansi().fg(Ansi.Color.YELLOW).a(logRecord.getLevel().getName()).reset().toString();
        }
        if (logRecord.getLevel().equals(Level.INFO)) {
            return Ansi.ansi().fgBright(Ansi.Color.BLUE).a(logRecord.getLevel().getName()).reset().toString();
        }
        return logRecord.getLevel().getName();
    }
}

File: cloudnet-app-lib/cloudnet-logging/src/main/kotlin/eu/cloudnetservice/cloudnet/v2/logging/LoggingFormatter.kt
package eu.cloudnetservice.cloudnet.v2.logging

import java.io.IOException
import java.io.PrintWriter
import java.io.StringWriter
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.logging.Formatter
import java.util.logging.LogRecord

class LoggingFormatter : Formatter() {
    private val dateFormat: DateFormat = SimpleDateFormat("dd.MM.yyyy HH:mm:ss")
    override fun format(record: LogRecord?): String {
        val builder = StringBuilder()
        if (record != null) {
            val message = formatMessage(record)
            builder.append('[')
                    .append(dateFormat.format(record.millis))
                    .append("] ")
                    .append(record.level.name)
                    .append(": ")
                    .append(message)

            if (!message.endsWith(System.lineSeparator())) {
                builder.append(System.lineSeparator())
            }

            if (record.thrown != null) {
                try {
                    StringWriter().use { writer ->
                        record.thrown.printStackTrace(PrintWriter(writer))
                        builder.append(writer).append(System.lineSeparator())
                    }
                } catch (e: IOException) {
                    e.printStackTrace()
                }
            }

        }

        return builder.toString()
    }
}